### PR TITLE
Cmake maloc

### PR DESCRIPTION
**Pull request by nsoblath**
Basic Maloc build works and has been integrated into the top-level FETK build

### CODE REVIEW

**Comment by nsoblath** [comment]
Closes #2

**Review by intendo — CHANGES_REQUESTED** [review]
@nsoblath I had no idea about the "Review Process" would not send you my comments until I submitted change requests.

File: maloc/src/config/maloccf.h.in [hunk 1]
```diff
@@ -62,118 +62,115 @@
 
 
 /* Does accept() use unsigned int? */
-#cmakedefine ACCEPT_USES_UINT
+#cmakedefine ACCEPT_USES_UINT 1
```
[the remark this answers is not in the document]
Comment (nsoblath) [reply maloc/src/config/maloccf.h.in:65]: I can certainly take the values out.  However, I may be missing something, but I don't see where in the documentation says that there shouldn't be values after the cmakedefine variables.  The page for the `configure_file` command says:

> Furthermore, input lines of the form
>   
> `#cmakedefine VAR ...`
>
> will be replaced with either
>   
> `#define VAR ...`
>
> or
>   
> /* #undef VAR */
> depending on whether VAR is set in CMake to any value not considered a false constant by the if() command. The "..." 
> content on the line after the variable name, if any, is processed as above. Input file lines of the form #cmakedefine01 VAR will be replaced with either #define VAR 1 or #define VAR 0 similarly.

It seems like the discussion of the "'...' content" is referring to values after the variable definition.

[the remark this answers is not in the document]
Comment (intendo) [reply maloc/src/config/maloccf.h.in:65]: @nsoblath ,
I think you should look at [CMakeChecks.basic-checks.cmake](https://www.ncbi.nlm.nih.gov/IEB/ToolBox/CPP_DOC/lxr/source/src/build-system/cmake/CMakeChecks.basic-checks.cmake) as an example on how to update `maloc/src/CMakeLists.txt`. 

Most of the `...` results can be defined by checking for values in `maloc/src/config/maloccf.h.in`. Adding the following to the `maloc/src/CMakeLists.txt` before the `configure_file` should fill in the correct `#define` value for `HAVE_ARPA_INET_H` in `maloc/src/config/maloccf.h.in` line 71 and put the correct result in the target `maloccf.h` file:
```
check_include_file(arpa/inet.h HAVE_ARPA_INET_H)
```
Likewise, for type sizes like line 68 in `maloc/src/config/maloccf.h.in`, you can add the following before the `configure_file` in `maloc/src/CMakeLists.txt`:
```
include(CheckTypeSize)
check_type_size("unsigned long" ACCEPT_USES_ULONG)
```

[the remark this answers is not in the document]
Comment (intendo) [reply maloc/src/config/maloccf.h.in:65]: @nsoblath we still need to get rid of all the `1`s in the #cmakedefine directives.

**Review by intendo — CHANGES_REQUESTED** [review]
@nsoblath we still need to get rid of all the 1s in the #cmakedefine directives.

**Comment by nsoblath** [comment]
@intendo I believe all of the requested changes and other things we've talked about have been addressed.  Please take a look and let me know.

**Review by intendo — APPROVED** [review]

**Pull request merged**